app/src/main/java/com/example/zingmp3phake/screen/mainapp/MainAppActivity.kt

```diff
@@ -19,25 +19,73 @@ import com.example.zingmp3phake.screen.personal.PersonalFragment
 import com.example.zingmp3phake.screen.search.SearchActivity
 import com.example.zingmp3phake.utils.MEDIA_EXTERNAL_AUDIO_URI
 import com.example.zingmp3phake.utils.base.BaseActivity
+import com.example.zingmp3phake.utils.loadByGlide
 
 class MainAppActivity :
     BaseActivity<ActivityMainAppBinding>(ActivityMainAppBinding::inflate),
     MainContract.View {
 
-    private val mainActivityPresenter = MainActivityPresenter(
-        SongRepository.getInstance(
-            LocalSong.getInstance(),
-            RemoteSong.getInstance()
-        ),
-        this
-    )
+    private lateinit var mainActivityPresenter: MainActivityPresenter
```

Comment:
```suggestion
    private lateinit var presenter: MainActivityPresenter
```
``presenter`` là đủ rồi em, vì mỗi Activity/ Fragment chỉ có 1 presenter.

Quan hệ giữa view & presenter là 1-1.